frontend/src/components/OrderForm/OrderForm.jsx

```diff
@@ -1,55 +1,257 @@
-import { useDispatch } from 'react-redux'
+import { useForm, useController } from 'react-hook-form'
+import { useSelector, useDispatch } from 'react-redux'
+import { useNavigate } from 'react-router-dom'
 import InputField from '../InputField/InputField'
-import './OrderForm.scss'
-import { useForm } from 'react-hook-form'
+import InputFieldDate from '../InputFieldDate/InputFieldDate'
+import Select from 'react-select'
+import { formOrderValidationSelectors } from '../../store/formOrderValidation/formOrderValidationSelectors'
 import { safeOrderForm } from '../../store/calculator/calculatorSlice'
-import { useNavigate } from 'react-router-dom'
-import { ROUTES } from '../../constants/constants'
-import { timeOptions } from '../../utils/initialData'
-
+import { ROUTES, customerStylesSelect } from '../../constants/constants'
+import { options } from '../../utils/initialData'
+import './OrderForm.scss'
 function OrderForm() {
-  const { register, handleSubmit } = useForm()
-  const dispath = useDispatch()
+  const stateDate = useSelector(formOrderValidationSelectors.getStateDate)
+  const dispatch = useDispatch()
+  const {
+    control,
+    reset,
+    register,
+    handleSubmit,
+
+    formState: { errors },
+  } = useForm({
+    mode: 'onBlur',
+  })
+
+  const required = 'Обязательное поле'
   const navigate = useNavigate()
   const onSubmit = data => {
-    dispath(safeOrderForm(data))
+    dispatch(safeOrderForm(data))
     navigate(ROUTES.payment)
+    reset()
   }
 
+  const { field } = useController({ name: 'cleaning_time', control })
+  const { value: slotValue, onChange: timeOnChange, ...restTimeField } = field
+
   return (
     <form onSubmit={handleSubmit(onSubmit)}>
-      <InputField isValid label="Имя" {...register('first_name')} />
-      <InputField isValid label="E-mail" placeholder="example@example.ru" {...register('email')} />
-      <InputField isValid label="Телефон" placeholder="+7 (999) 999-99-99" {...register('phone')} />
-      <InputField isValid label="Город" {...register('city')} />
-      <InputField isValid label="Улица" {...register('street')} />
+      {/* -------------------------------------USERNAME--------------------------------- */}
+      <div className="inputs_wrapper-field">
+        <InputField
+          isValid
+          label="Имя"
+          {...register('username', {
+            required,
+            maxLength: {
+              value: 60,
+              message: 'Максимум 60 символов',
+            },
+            pattern: /^(?=.{1,40}$)[а-яёА-ЯЁ]+$/,
```

Comment:
Надо для паттерна добавить везде сообщение об ошибке, чтобы пользователю было очевидно, что нужно исправить)
```suggestion
            pattern: {
              value: /^(?=.{1,40}$)[а-яёА-ЯЁ]+$/,
              message: 'Укажите ваше имя. Пример: Апполинарий Вальдемарович фон Спасо-Преображенский',
            },
```